.github/workflows/publish.yaml

```
@@ -0,0 +1,90 @@
+name: Publish packages on Nuget
+
+on:
+  workflow_dispatch:
+    inputs:
+      publish-package:
+        description: "Build or Build and Publish?"
+        required: true
+        type: choice
+        default: "Build"
+        options:
+          - "Build"
+          - "Build and Publish"
+
+permissions: {}
+
+jobs:
+  downloadbinaries:
+    runs-on: ubuntu-latest
+    outputs:
+      package-version: ${{ steps.package.outputs.package-version }}
+      serverless-compat-version: ${{ steps.serverlesscompatbinary.outputs.serverless-compat-version }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - id: package
+        run: |
+          PACKAGE_VERSION=$(awk -F '[<>]' '/<Version>/{print $3; exit}' Datadog.Serverless.Compat/Datadog.Serverless.Compat.csproj)
+          echo "package-version=$PACKAGE_VERSION" >> "$GITHUB_OUTPUT"
+      - id: serverlesscompatbinary
+        run: |
+          RESPONSE=$(curl -s "https://api.github.com/repos/datadog/serverless-components/releases")
+          SERVERLESS_COMPAT_VERSION=$(echo "$RESPONSE" | jq -r --arg pattern "datadog-serverless-compat\/v[0-9]*\.[0-9]*\.[0-9]*" '.[] | select(.tag_name | test($pattern)) | .tag_name' | sort -V | tail -n 1)
+
+          echo "Using version ${SERVERLESS_COMPAT_VERSION} of Serverless Compatibility Layer binary"
+          echo "serverless-compat-version=$(echo "$SERVERLESS_COMPAT_VERSION" | jq -rR 'ltrimstr("sls-")')" >> "$GITHUB_OUTPUT"
+
+          curl --output-dir ./temp/ --create-dirs -O -s -L "https://github.com/DataDog/serverless-components/releases/download/${SERVERLESS_COMPAT_VERSION}/datadog-serverless-compat.zip"
+          unzip ./temp/datadog-serverless-compat.zip -d ./Datadog.Serverless.Compat/datadog
+      - uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
+        with:
+          name: bin
+          path: Datadog.Serverless.Compat/datadog/bin
+  build:
+    needs: [downloadbinaries]
+    runs-on: ubuntu-latest
+    env:
+      PACKAGE_VERSION: ${{ needs.downloadbinaries.outputs.package-version }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          path: Datadog.Serverless.Compat/datadog
+      - uses: actions/setup-dotnet@67a3573c9a986a3f9c594539f4ab511d57bb3ce9 #4.3.1
+        with:
+          dotnet-version: "6.0.x"
+      - run: cd Datadog.Serverless.Compat && dotnet build -c Release && dotnet pack -c Release
```

`dotnet pack` will impolitely build the project, so no need to call `dotnet build` separately.
```suggestion
      - run: cd Datadog.Serverless.Compat && dotnet pack -c Release
```